### ISSUE TITLE
sp_DatabaseRestore:  call sp_CommandLog for additional statements

### ISSUE THREAD

**Issue by jackmang** [BrentOzarULTD/SQL-Server-First-Responder-Kit#1436]
I'm not sure if this is a bugfix or a feature, but currently some of the key executed script items are not run through sp_CommandExecute, and therefore are not in the CommandLog table.

Specifically, these are the changes I suggest:
Change execution of the final RESTORE DATABASE WITH RECOVERY to use CommandExecute instead of sp_executesql.
Change execution of ALTER DATABASE SET RECOVERY SIMPLE to use CommandExecute instead of sp_executesql.
Change execution of EXECUTE [dbo].[DatabaseIntegrityCheck] to use CommandExecute instead of sp_executesql.
Change execution of DROP DATABASE to use CommandExecute instead of sp_executesql.

I prematurely created a pull request that included these changes #1426

**Comment by BrentOzar** [comment]
This all makes sense, and I like it. 

When you do another pull request for it though, make sure the pull request only works on one particular feature at a time. I did a quick read through that pull request, and a lot of the code was unrelated (like debug checks). Keeping your code focused on one issue at a time just means it's easier to coexist with other developers working on other features.

**Comment by jackmang** [comment]
Agreed.  We'll try to do this incrementally.

**Comment by jackmang** [comment]
So, I'm trying to learn the right way to contribute.  I think that before I submit the pull request I need to get agreement on what it looks like.  It is pretty simple.  I will replace the last 4 instances of
`EXECUTE sp_executesql @sql;`
with  
`EXECUTE @sql = [dbo].[CommandExecute] @Command = @sql, @CommandType = 'XXXXXXXX', @Mode = 1, @DatabaseName = @Database, @LogToTable = 'Y', @Execute = 'Y';`

while adjusting the 'XXXXXXXXX' with and appropriate summary of the command type for the sql being executed.

```
SET @sql = N'RESTORE DATABASE ' + @RestoreDatabaseName + N' WITH RECOVERY' + NCHAR(13);
...
EXECUTE @sql = [dbo].[CommandExecute] @Command = @sql, @CommandType = 'RESTORE DATABASE', @Mode = 1, @DatabaseName = @Database, @LogToTable = 'Y', @Execute = 'Y';
```

```
SET @sql = N'ALTER DATABASE ' + @RestoreDatabaseName + N' SET RECOVERY SIMPLE' + NCHAR(13);
...
EXECUTE @sql = [dbo].[CommandExecute] @Command = @sql, @CommandType = 'ALTER DATABASE', @Mode = 1, @DatabaseName = @Database, @LogToTable = 'Y', @Execute = 'Y';

```
```
SET @sql = N'EXECUTE [dbo].[DatabaseIntegrityCheck] @Databases = ' + @RestoreDatabaseName + N', @LogToTable = ''Y''' + NCHAR(13);
...
EXECUTE @sql = [dbo].[CommandExecute] @Command = @sql, @CommandType = 'INTEGRITY CHECK', @Mode = 1, @DatabaseName = @Database, @LogToTable = 'Y', @Execute = 'Y';

```
```
SET @sql = N'DROP DATABASE ' + @RestoreDatabaseName + NCHAR(13);
...
EXECUTE @sql = [dbo].[CommandExecute] @Command = @sql, @CommandType = 'DROP DATABASE', @Mode = 1, @DatabaseName = @Database, @LogToTable = 'Y', @Execute = 'Y';
```

On a side note, it seems odd that the script is terminating each command with NCHAR(13).  I presume that this is to format the output, but it doesn't feel right.  I'll plan to leave that alone for now.

**Comment by BrentOzar** [comment]
OK, sounds good. I've never used CommandExecute, so I leave that part in your capable hands. ;-)

**Comment by BlitzErik** [comment]
Closed via #1463
